AdobeStockImage/Model/AssetIndexer.php

```diff
@@ -0,0 +1,164 @@
+     */
+    private $service;
+
+    /**
+     * @var GetByPathInterface
+     */
+    private $getByPathCommand;
+
+    /**
+     * @var AssetRepositoryInterface
+     */
+    private $assetRepository;
+
+    /**
+     * @var SearchCriteriaBuilder
+     */
+    private $searchCriteriaBuilder;
+
+    /**
+     * @var Filesystem
+     */
+    private $filesystem;
+
+    /**
+     * @var ReadInterface
+     */
+    private $mediaDirectory;
+
+    /**
+     * @var File
+     */
+    private $driver;
+
+    /**
+     * @var ResourceConnection
+     */
+    private $resource;
+
+    /**
+     * Constructor
+     *
+     * @param GetByPathInterface $getByPathCommand
+     * @param ResourceConnection $resource
+     * @param AssetRepositoryInterface $assetRepository
+     * @param SearchCriteriaBuilder $searchCriteriaBuilder
+     * @param Filesystem $filesystem
+     * @param Service $service
+     * @param File $driver
+     */
+    public function __construct(
+        GetByPathInterface $getByPathCommand,
+        ResourceConnection $resource,
+        AssetRepositoryInterface $assetRepository,
+        SearchCriteriaBuilder $searchCriteriaBuilder,
+        Filesystem $filesystem,
+        Service $service,
+        File $driver
+    ) {
+        $this->getByPathCommand = $getByPathCommand;
+        $this->assetRepository = $assetRepository;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+        $this->filesystem = $filesystem;
+        $this->service = $service;
+        $this->driver = $driver;
+        $this->resource = $resource;
+    }
+
+    /**
+     * Check if image licensed and save it to media gallery asset grid
+     *
+     * @param \SplFileInfo $item
+     * @return void
+     */
+    public function execute(\SplFileInfo $item): void
+    {
+        $mediaAsset = $this->getByPathCommand->execute($this->getPathFormMediaAsset($item));
+
+        $this->searchCriteriaBuilder->addFilter('media_gallery_id', $mediaAsset->getId());
+        $searchCriteria = $this->searchCriteriaBuilder->create();
+
+        $assets = $this->assetRepository->getList($searchCriteria)->getItems();
+
+        if (count($assets) > 0) {
+            foreach ($assets as $asset) {
+                $this->getConnection()->insertOnDuplicate(
+                    $this->resource->getTableName('media_gallery_asset_grid'),
+                    [
+                        'id' => $mediaAsset->getId(),
+                        'licensed' => $asset->getIsLicensed(),
+                    ]
+                );
+            }
+        }
+    }
+
+    /**
+     * Return correct path for file.
+     *
+     * @param \SplFileInfo $item
+     * @return string
+     */
+    private function getPathFormMediaAsset(\SplFileInfo $item): string
+    {
+        $path = $this->getMediaDirectory()->getRelativePath($item->getPath() . '/' . $item->getFileName());
+
+        if ($this->driver->getParentDirectory($path) === '.') {
+            $path = '/' . $path;
```

Comment:
It's not clear to me why do we need this transformation, can you please provide an example